lib/sidekiq/form526_backup_submission_process/processor.rb

```diff
@@ -428,29 +430,39 @@ def convert_doc_to_pdf(doc, klass)
         end
       end
 
-      def choose_provider(headers, breakered: true)
+      # 82245 - Adding provider to method. this should be removed when toxic exposure flipper is removed
+      def choose_provider(headers, provider, breakered: true)
```

Comment:
Is this the only method that needs to be made public, or was there a different reason for making all methods within this class accessible?

Reply:
fair questions. @SamStuckey do you have any issue with me moving this method to private? I dont know if you intend to use it in the remediation process

Reply:
Ah yes... So basically I did the bad thing and made a private method public so i could test it. I feel ok with doing this because the question of private / public methods and what you test is kind of built on the idea that the object in question is built using OOD best practice, which this one is not IMHO. As I see it our options are
1. refactor the object into smaller objects that do one thing, each with a predicable API and test accordingly. 
2. Keep the method private and remove the test
3. keep the method private and write a test that stubs out everything else this does with the aim of testing this.

Honestly, given the use case of this object and how cobbled together it is anyway, I figured erring on the side of an extra test was preferable, but ultimately it's a shortcut

Reply:
I'm leaving it alone for now, and adding a tech debt story to clean things up.